save-cloud-charts/save-cloud/templates/backend-deployment.yaml

```diff
@@ -18,6 +18,7 @@ spec:
       annotations:
         {{- include "pod.common.annotations" (dict "service" .Values.backend ) | nindent 8 }}
     spec:
+      serviceAccountName: microservice-sa
```

Comment:
As far as I can see, all the microservices that are able to send requests should have `microservice-sa` (e.g. `demo-cpg` does not send anything so service account for it is not required(?))

Moreover, it seems that orchestrator and demo role bindings should also reference `microservice` cluster role

Reply:
The idea was, that communication between some microservices can still be secure without security, because we were planning to account only for not trusted executed tools and their network activity could be restricted with a `NetworkPolicy` on agent pods. This way using SA token for authentication is only required for microservices which receive requests from agents. But then, microservices that send requests to these services also need to authenticate, so it may be easier to unify everything.

Reply:
As far as I can see, many things have been done: `KubernetesAuthenticaitonUtils` provide almost everything for `X-Service-Account-Token` check. So the question is: what hasn't been done yet? (regarding this PR) Seems that ktor needs to be able to add custom headers. And it seems that some `Bean`s are required (e.g. `KubernetesClient` bean in `KubernetesAuthenticationUtils`). Am I right?

Reply:
I suppose it was already more or less working. This PR affects only services, not agents, and I'm not sure if we have ktor on server-side. But yes, the part that validates token against k8s api server requires `KubernetesClient`. The part that adds it as a custom header requires the token to be mounted (added to YAML spec as `projected volume`)